prac_07/project_management.py

```diff
@@ -0,0 +1,113 @@
+"""
+Estimated: 4hrs
+Actual:
+"""
+from prac_07.project import Project
+import datetime
+
+
+def main():
+    # filename = "projects.txt"
+    projects = []
+    print_menu()
+    user_input = input(">>> ").lower()
+    while user_input != 'q':
+        if user_input == 'l':
+            filename = input("Enter Filename: ")
+            projects = []
+            read_data(filename, projects)
+
+        elif user_input == 's':
+            write_to_file(projects)
+
+        elif user_input == 'd':
+            completed_projects = []
+            incomplete_projects = []
+            for project in projects:
+                if project.completion_percentage < 100:
+                    incomplete_projects.append(project)
+                else:
+                    completed_projects.append(project)
+            print_projects(completed_projects, incomplete_projects)
+
+        elif user_input == 'f':  # Incomplete
+            date_string = input("Show projects that start after date (dd/mm/yy): ")
+            date = datetime.datetime.strptime(date_string, "%d/%m/%Y").date()
+            filtered_projects = [project for project in projects if project.start_date > date]
+            for project in filtered_projects:
+                print(project)
+
+        elif user_input == 'a':
+            print("Let's add a new project")
+            add_project(projects)
+
+        elif user_input == 'u':
+            for i, project in enumerate(projects, 0):
+                print(f"{i} {project}")
+            update_project(projects)
+        else:
+            print("Error")
+        print_menu()
+        user_input = input(">>> ").lower()
+
+
+def update_project(projects):
```

Comment:
It's missing updating the priority as well. It also needs to not change the value if an empty string is given.